static/panes/tool-input-view.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2021, Tom Ritter
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import * as monaco from 'monaco-editor';
+import _ from 'underscore';
+import { MonacoPane } from './pane';
+import { ga } from '../analytics';
+import * as monacoConfig from '../monaco-config';
+import { Container } from 'golden-layout';
+import { MonacoPaneState } from './pane.interfaces';
+import { Hub } from '../hub';
+import { ToolInputViewState } from './tool-input-view.interfaces';
+import { Settings } from '../settings';
+
+export class ToolInputView extends MonacoPane<monaco.editor.IStandaloneCodeEditor, ToolInputViewState> {
+    source: string;
+    _toolId: number;
+    _toolName: string;
+    shouldSetSelectionInitially: boolean;
+    debouncedEmitChange: (() => void) & _.Cancelable;
+    cursorSelectionThrottledFunction: ((e: any) => void) & _.Cancelable;
+    lastChangeEmitted: string | undefined;
+    constructor(hub: Hub, container: Container, state: ToolInputViewState & MonacoPaneState) {
+        state = state || {}; // TODO according to TS typing this will be a no-op
```

Comment:
If state might me undefined/null, shouldn't that be stated in the type? Either `state?` or adding ` | null`

Reply:
Any thoughts on this @jeremy-rifkin ?

Reply:
Pardon my delayed reply, I'll make it `state?`. I'm not sure if a null/undefined state can be passed in anymore but I figure it's best to leave this check just in case, at least for now.